import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java

```diff
@@ -405,4 +643,407 @@ private String getPropertieFieldName(String lang,String langPackType) {
+    }
+
+    protected void processLocationElementType(InternalDocument document, Element element) {
+        Consumer<Field> locationConsumer = field -> {
+            NodeList fieldNodeList = element.getElementsByTagName(field.getFieldName());
+            List<String> codesLocation = new ArrayList<>();
+            for (int j = 0; j < fieldNodeList.getLength(); j++) {
+                Element fieldElement = (Element) fieldNodeList.item(j);
+                String name = getNameFromElement(fieldElement);
+
+                if (!StringUtils.isBlank(name)) {
+                    codesLocation.add(name);
+                    FieldValue fv = new FieldValue();
+                    if (name != null && !name.isEmpty()) {
+                        fv.setCode(name);
+                        fv.setValue(name);
+                        fv.setSelected(true);
+                    }
+                    int index = field.getPossibleValues() == null ? 0 : field.getPossibleValues().size();
+                    fv.setIndex(index);
+                    if (field.getPossibleValues() == null) {
+                        field.setPossibleValues(new ArrayList<FieldValue>());
+                    }
+                    if (!field.getPossibleValues().stream().anyMatch(n -> {
+                        return n.getCode().equals(fv.getValue());
+                    })) {
+                        field.getPossibleValues().add(fv);
+                    }
+                }
+                if (!codesLocation.isEmpty()) {
+                    String[] codeValues = codesLocation.stream().toArray(String[]::new);
+                    document.addStringMultiField(field.getFieldName(), codeValues);
+                }
+            }
+        };
+        processForEachFilteredByType(locationConsumer, FieldType.LOCATION);
+    }
+
+    protected void processDateElementType(InternalDocument document, Element element) throws ParseException {
+        NodeList nodes = element.getElementsByTagName("activity-date");
+        for (int j = 0; j < nodes.getLength(); ++j) {
+            Element e = (Element) nodes.item(j);
+            Field field = getField(e, FieldType.DATE);
+            String localDate = e.getAttribute("iso-date");
+                String format = "yyyy-MM-dd";
+                SimpleDateFormat sdf = new SimpleDateFormat(format);
+                if (localDate != null && !localDate.isEmpty()) {
+                    document.addDateField(field.getUniqueFieldName(), sdf.parse(localDate));
+                }
+        }
+    }
+
+
+    protected void processTransactionElementType(InternalDocument document, Element iatiActivity,
+                                                 Element activityProviderNode)  {
+        NodeList nodes;
+        nodes = iatiActivity.getElementsByTagName("transaction");
+        for (int j = 0; j < nodes.getLength(); ++j) {
+            String reference ;
+            Element e = (Element) nodes.item(j);
```

Comment:
Rename e to something more descriptive e.g transactionElement

Reply:
done